improver/psychrometric_calculations/total_precipitable_water.py

```diff
@@ -0,0 +1,147 @@
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module to generate total precipitable water"""
+
+import iris
+import numpy as np
+from iris.cube import Cube
+from numpy import ndarray
+
+from improver import BasePlugin
+from improver.constants import EARTH_SURFACE_GRAVITY_ACCELERATION, WATER_DENSITY
+
+
+def ensure_no_float_64(cube: Cube) -> Cube:
+    """
+    Convert float64 data and coordinates in the cube to float32 to ensure workflow compatibility.
+    """
+    if cube.data.dtype == np.float64:
+        cube.data = cube.data.astype(np.float32)
+    for coord in cube.coords():
+        if coord.points.dtype == np.float64:
+            coord.points = coord.points.astype(np.float32)
+        if coord.bounds is not None and coord.bounds.dtype == np.float64:
+            coord.bounds = coord.bounds.astype(np.float32)
+    return cube
+
+
+class PrecipitableWater(BasePlugin):
+    """
+    Plugin to calculate total precipitable water from a humidity mixing ratio cube.
+
+    This plugin integrates the humidity mixing ratio over pressure levels to compute
+    the total precipitable water (TPW) in metres. It assumes pressure levels are in
+    Pascals and uses constants for gravity and water density to convert the result
+    into liquid water equivalent thickness.
+    """
+
+    @staticmethod
+    def calculate_pressure_thickness(
+        layer_bounds: ndarray, cube_data: ndarray
+    ) -> ndarray:
+        """
+        Calculate the pressure thickness of each atmospheric layer.
+
+        Args:
+            layer_bounds:
+                1D array of pressure layer boundaries in Pascals.
+            cube_data:
+                N-dimensional array of humidity mixing ratio values.
+
+        Returns:
+            N-dimensional array of pressure thickness values reshaped to match
+            the dimensions of the input data.
+        """
+        delta_p = np.abs(np.diff(layer_bounds))
+        reshaped_delta_p = delta_p.reshape(
+            (len(delta_p),) + (1,) * (cube_data.ndim - 1)
+        )
+        return reshaped_delta_p
+
+    @staticmethod
+    def calculate_precipitable_water(
+        sorted_data: ndarray, reshaped_delta_p: ndarray
+    ) -> ndarray:
+        """
+        Calculate total precipitable water using sorted humidity data and pressure thickness.
+
+        Args:
+            sorted_data:
+                Humidity mixing ratio data sorted by pressure levels.
+            reshaped_delta_p:
+                Pressure thickness values reshaped to match the data dimensions.
+
+        Returns:
+            Array of total precipitable water values in metres.
+        """
+        pw_data = (
+            sorted_data
+            * reshaped_delta_p
+            / (EARTH_SURFACE_GRAVITY_ACCELERATION * WATER_DENSITY)
+        )
+        total_pw_data = np.maximum(pw_data, 0)
```

Comment:
I think the purpose of this statement is to ensure that the precipitable water values are not negative. This could only happen if either the humidity values are negative or the pressure values are negative. The `np.abs()` in the pressure thickness method means that the pressure values can never be negative, and negative humidity mixing ratio values are unphysical. 
With this in mind, I think this statement can be removed.